### PR TITLE
Add owner_name column for table repository for maintaince reason

### PR DESCRIPTION
**Pull request by lunny**
As title. Inspired by @zeripath 's comment.

### CODE REVIEW

File: models/repo.go [hunk 1]
```diff
@@ -149,9 +149,9 @@ const (
 
 // Repository represents a git repository.
 type Repository struct {
-	ID                  int64                  `xorm:"pk autoincr"`
-	OwnerID             int64                  `xorm:"UNIQUE(s) index"`
-	OwnerName           string                 `xorm:"-"`
+	ID                  int64 `xorm:"pk autoincr"`
+	OwnerID             int64 `xorm:"UNIQUE(s) index"`
+	OwnerName           string
```
Comment (guillep2k) [inline models/repo.go:154]: ~~Perhaps index this column, so input like "xorm/core" can be queried just with the `repository` table?~~

I see it's `Name` and not `LowerName`. Boomer. 😔

File: modules/context/repo.go [hunk 2]
```diff
@@ -256,7 +256,7 @@ func RedirectToRepo(ctx *Context, redirectRepoID int64) {
 	redirectPath := strings.Replace(
 		ctx.Req.URL.Path,
 		fmt.Sprintf("%s/%s", ownerName, previousRepoName),
-		fmt.Sprintf("%s/%s", repo.MustOwnerName(), repo.Name),
+		fmt.Sprintf("%s", repo.FullName()),
```
Comment (guillep2k) [inline modules/context/repo.go]:
```suggestion
		repo.FullName(),
```

**Review by guillep2k — APPROVED** [review]
This should improve the speed quite a bit.

Just a nit (and the CI that's acting up).

File: models/repo.go (hunk 1 above)
Comment (lunny) [reply models/repo.go:154]: This column shouldn't be index, because we have `owner_id`. This column is a duplicate column only for easy repopath.

**Comment by zeripath** [comment]
I don't see any code changing the owner name during transfer repository - have you covered this case?

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
You need to update the test fixtures

**Comment by zeripath** [comment]
In particular this file:

**Comment by zeripath** [comment]
Ok so it looks like there are a few places where owner is expected to be present because it would have been loaded in order to get the owner name. I'm not sure what's best here, it might be that it's easiest to just stick LoadOwner s before every call to repo. Owner but that could easily cause some deadlocks - and that still doesn't necessarily catch the templates.

There's also an issue with a unit test.

**Comment by zeripath** [comment]
We should expect to have to fix several NPEs in the coming weeks regarding this - however, I still think this is worthwhile. 

We can also begin the process of removing all of the unnecessary loadOwner calls that this will free us from making.

**Pull request merged**